software/speller/data_collection_platform/dcp/api.py

```diff
@@ -38,38 +39,128 @@ def index():
     # TODO return keyboard layout
     return ' '
 
-
-@bp.route('/char_pressed', methods=['POST', 'GET'])
-def char_pressed():
-    if request.method == 'POST':
-        user_char_pressed = request.form['char_pressed']
-
-
 @bp.route("/openbci/start", methods=['POST'])
 def openbci_start():
     # use a separate process to stream BCI data
     from dcp.bci.stream import stream_bci_api
     from multiprocessing import Process
     with current_app.app_context():
-        p = Process(target=stream_bci_api, args=(shared.bci_processes_states,))
+        # Although the subprocess gets initialized inside this scope, it will persist
+        # bci_processes_states references this subprocess_dict
+        subprocess_dict = shared.manager.dict({
+            'character': None,
+            'phase': None,
+            'frequency': None,
+            'q': deque(),
```

Comment:
I think we can't use a deque here, because it is not multithreading safe. Instead we should using `from multiprocessing import Queue` see https://docs.python.org/3/library/queue.html#queue.Queue.

Reply:
i think that objects in a manager.dict dictionary can only be accessed in a thread safe manner

Reply:
![image](https://user-images.githubusercontent.com/43357040/149645533-e67f4cf6-a586-4497-b11d-a6b0f1380669.png)

I think in this case should be initialize as `manager.Queue()` according to the docs